Fix and refactor Langfuse Terraform deployment for EKS compatibility

ahmedbaig229 opened the pull request:
- Fixed deprecated AWS region attribute in ALB Controller.
- Updated Langfuse Helm release to use a dynamically created namespace.
- Improved subnet validation using coalesce() for safer variable evaluation.
- Added provider blocks for AWS SSO and EKS cluster connectivity.

Tested deployment successfully on AWS us-east-1 (EKS Fargate). Verified Helm release and ingress routing with HTTPS and DNS hostname.

ahmedbaig229 committed:
fix(helm): update aws_load_balancer_controller region field from .name to .id (deprecated attr)

ahmedbaig229 committed:
fix(terraform): refactor helm_release for langfuse to use dynamic namespace and dependency order

ahmedbaig229 committed:
fix(validation): use coalesce for subnet validation to avoid null length errors

ahmedbaig229 committed:
feat(providers): add aws, kubernetes, and helm provider configurations for SSO profile and cluster integration

ahmedbaig229 committed:
Update README.md

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/langfuse/langfuse-terraform-aws?pullRequest=36) <br/>All committers have signed the CLA.

Steffen911 commented on README.md:
```diff
@@ -338,3 +338,15 @@ This module creates a complete Langfuse stack with the following components:
+- **Subnet Validation:** Used `coalesce` to prevent null length errors when VPC exists.
+- **Providers:** Added AWS SSO profile and integrated Kubernetes & Helm providers for EKS cluster.
+
+> All commits authored by Ahmed Ali Baig using GitHub noreply email to keep private information safe.
```
Please remove this.

ahmedbaig229 replied:
I have remove these from README file, please verify it thanks

Steffen911 commented on versions.tf:
```diff
@@ -28,3 +28,23 @@ terraform {
+provider "aws" {
+  profile = "di"
+  region  = "us-east-1"
+}
```
What is the purpose of the records here?

ahmedbaig229 replied:
Basically, in my case, I use this provider block to deploy into my client’s account. As per the best security practices in my organization, we don’t use AWS user access keys and secret keys in client accounts as permanent credentials. Sometimes, these keys can be easily exposed due to improper use — for example, by junior developers who might not be aware and treat them as normal. To follow best practices, we use AWS cross-account roles to access client accounts. For that, I use AWS CLI SSO login to switch from my account to the client’s account.

Steffen911 reviewed (commented):
Thank you for the contribution! Please explain the reasons for changing the files with comments and the purposes of those edits.

ahmedbaig229 committed:
Remove changelog

Removed changelog section for Terraform Langfuse Deployment Fix.

ahmedbaig229 committed:
Comment out AWS provider settings in versions.tf

Steffen911 commented on versions.tf:
```diff
@@ -28,3 +28,26 @@ terraform {
+#provider "aws" {
+#  profile = "di"
+#  region  = "us-east-1"
+#}
```
Could we remove this if it's not being used by the module? Same with the kubernetes settings below if they are not in use.

ahmedbaig229 replied:
Hi Steffen, as a DevSecOps Engineer, I recommend to use this as an entry point instead of using the access key and private key, and it should be here, I mistakenly comments those blocks, but if you want i can remove it, thanks

ahmedbaig229 replied:
These providers are needed for Helm and Kubernetes resources to connect to the EKS cluster the helm_release uses them internally to deploy Langfuse and the AWS Load Balancer Controller into the cluster. If we don’t define these providers, Terraform can’t connect to the EKS cluster we will get errors like:
“Error: Kubernetes cluster unreachable” or “provider configuration not present” when running terraform apply for Helm or Kubernetes resources.

Steffen911 replied:
@ahmedbaig229 Yes, they are necessary and that's why we document them in our usage guide within the readme: https://github.com/langfuse/langfuse-terraform-aws?tab=readme-ov-file#usage. From my perspective, they should live outside of the module though and don't be declared within the versions.tf file.

Can you explain why we would want those declarations within the module?

ahmedbaig229 replied:
Yeah, these providers do work and are required for Helm/Kubernetes resources. The only reason to include them inside the module is to make the module self-contained so it can immediately connect to the EKS cluster it creates, removing the need for users to configure provider blocks themselves. Otherwise, Terraform fails with ‘cluster unreachable’ or missing provider configs. But following Terraform best practices, it’s cleaner to keep provider declarations outside the module, so I’m fine removing them.

Steffen911 commented on langfuse.tf:
```diff
@@ -197,12 +197,11 @@ resource "kubernetes_secret" "langfuse" {
+  repository = "https://langfuse.github.io/langfuse-k8s"
+  version    = var.langfuse_helm_chart_version
+  chart      = "langfuse"
+  namespace  = kubernetes_namespace.langfuse.metadata[0].name
```
Just to confirm - currently the namespace is still hard-coded to langfuse in the kubernetes_namespace declaration. Do you want to make this configurable?

ahmedbaig229 replied:
Yeah, we can change via a variable, but it's fine as long we use this as a it as a single environment

ahmedbaig229 committed:
Clean up versions.tf by removing Provider blocks

Removed commented AWS SSO profile configuration and redundant provider settings.
